pkg/gadgets/run/types/metadata.go

```diff
@@ -395,3 +447,93 @@ func (m *GadgetMetadata) populateStruct(btfStruct *btf.Struct) error {
 
 	return nil
 }
+
+func (m *GadgetMetadata) populateParams(spec *ebpf.CollectionSpec) error {
+	var result error
+
+	paramNames, err := getGadgetIdentByPrefix(spec, paramPrefix)
+	if err != nil {
+		result = multierror.Append(result, err)
+	}
+
+	for _, name := range paramNames {
+		var btfVar *btf.Var
+		err := spec.Types.TypeByName(name, &btfVar)
+		if err != nil {
+			result = multierror.Append(result, fmt.Errorf("variable %q not found in eBPF object: %w", name, err))
+			continue
+		}
+
+		err = checkParamVar(spec, name)
+		if err != nil {
+			result = multierror.Append(result, err)
+			continue
+		}
+		_, found := m.EBPFParams[name]
+		if found {
+			log.Debugf("Param %q already defined, skipping", name)
+			continue
+		}
+
+		log.Debugf("Adding param %q", name)
+		m.EBPFParams[name] = EBPFParam{
+			ParamDesc: params.ParamDesc{
+				Key:         name,
+				Description: "TODO: Fill parameter description",
+			},
+		}
+	}
+
+	return result
+}
+
+func checkMacroParamVar(btfVar *btf.Var) error {
+	var result error
+
+	if btfVar.Linkage != btf.GlobalVar {
+		result = multierror.Append(result, fmt.Errorf("%q is not a global variable", btfVar.Name))
+	}
+	btfPtr, ok := btfVar.Type.(*btf.Pointer)
+	if !ok {
+		result = multierror.Append(result, fmt.Errorf("%q is not a pointer", btfVar.Name))
+		return result
+	}
+	btfConst, ok := btfPtr.Target.(*btf.Const)
+	if !ok {
+		result = multierror.Append(result, fmt.Errorf("%q is not const", btfVar.Name))
+		return result
+	}
+	_, ok = btfConst.Type.(*btf.Void)
+	if !ok {
+		result = multierror.Append(result, fmt.Errorf("%q is not a const void pointer", btfVar.Name))
+		return result
+	}
+
+	return result
+}
```

Comment:
IMO all this can be moved to getGadgetIdentByPrefix. (Otherwise the name needs to be changed as it's not only about params anymore.